components/ai_chat/renderer/page_text_distilling.cc

```diff
@@ -204,4 +225,102 @@ void DistillPageText(
   std::move(callback).Run(contents_text);
 }
 
+void DistillPageTextViaSiteScript(
+    content::RenderFrame* render_frame,
+    const std::string& script_content,
+    int32_t world_id,
+    base::OnceCallback<void(const std::optional<std::string>&)> callback) {
+  /**
+   * Concatenates our injector-script (retrieved from the resource bundle) with
+   * our extractor script. Because the injector-script is wrapped in an IIFE, we
```

Comment:
can't we export a function that takes a handler? Then we can wrap the whole and it will be completely self-contained.

Reply:
As discussed on Slack, there was a seemingly-unshakable IIFE that would be generated with the script output. I was able to find a way to remove that, and now working on an approach that will be self-contained.

Reply:
@bridiver, just pushed a new approach involving no events.